x-pack/legacy/plugins/ml/public/overview/components/anomaly_detection_panel/anomaly_detection_panel.tsx

```diff
@@ -60,59 +52,56 @@ function getDefaultAnomalyScores(groups: Group[]): MaxScoresByGroup {
 
 export const AnomalyDetectionPanel: FC = () => {
   const [isLoading, setIsLoading] = useState(false);
-  const [groups, setGroups] = useState<Groups>({});
+  const [groups, setGroups] = useState<GroupsDictionary>({});
   const [groupsCount, setGroupsCount] = useState<number>(0);
-  const [jobsList, setJobsList] = useState<Jobs>([]);
+  const [jobsList, setJobsList] = useState<MlSummaryJobs>([]);
   const [statsBarData, setStatsBarData] = useState<any>(undefined);
   const [errorMessage, setErrorMessage] = useState<any>(undefined);
 
   const loadJobs = async () => {
     setIsLoading(true);
 
     try {
-      const jobsResult: Jobs | any = await ml.jobs.jobsSummary([]);
-      const jobsSummaryList = jobsResult.map((job: any) => {
+      const jobsResult: MlSummaryJobs = await ml.jobs.jobsSummary([]);
+      const jobsSummaryList = jobsResult.map((job: MlSummaryJob) => {
         job.latestTimestampSortValue = job.latestTimestampMs || 0;
         return job;
       });
-      const jobsGroups = getGroupsFromJobs(jobsSummaryList);
+      const { groups: jobsGroups, count } = getGroupsFromJobs(jobsSummaryList);
       const jobsWithTimerange = getJobsWithTimerange(jobsSummaryList);
       const stats = getStatsBarData(jobsSummaryList);
       setIsLoading(false);
       setErrorMessage(undefined);
       setStatsBarData(stats);
-      setGroupsCount(Object.values(jobsGroups).length);
+      setGroupsCount(count);
       setGroups(jobsGroups);
       setJobsList(jobsWithTimerange);
       loadMaxAnomalyScores(jobsGroups);
     } catch (e) {
-      setErrorMessage(JSON.stringify(e));
+      setErrorMessage(e.message !== undefined ? e.message : JSON.stringify(e));
       setIsLoading(false);
     }
   };
 
-  const loadMaxAnomalyScores = async (groupsObject: Groups) => {
+  const loadMaxAnomalyScores = async (groupsObject: GroupsDictionary) => {
     const groupsList: Group[] = Object.values(groupsObject);
     const scores = getDefaultAnomalyScores(groupsList);
 
     try {
-      const promises: any[] = [];
-      groupsList.forEach((group, i) => {
-        if (group.jobIds.length > 0) {
+      const promises = groupsList
+        .filter(group => group.jobIds.length > 0)
+        .map((group, i) => {
           scores[group.id].index = i;
-          promises.push(
-            mlResultsService.getOverallBucketScores(
-              group.jobIds,
-              1,
-              group.earliest_timestamp,
-              group.latest_timestamp
-            )
+          return mlResultsService.getOverallBucketScores(
```

Comment:
Since we are only displaying max bucket score for now, I'd switch this for a new call which just returns the maximum `anomaly_score` for `result_type:bucket` over all the jobs in the group. When you switch the contents of this cell to a swimlane, you will need to pass in an `interval`, which will depend on the width of the swimlane in the cell and the number of buckets you want to display - similar to `getSwimlaneBucketInterval` in `explorer.js`.